### PR TITLE
Set up webpack configuration for building examples

### PR DESCRIPTION
**Pull request by schneidmaster**
Use webpack to build examples in development and prod. Faster rebuilds and a nicer dev experience.

### CODE REVIEW

File: examples/App.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import React from 'react'
```
Comment (ianstormtaylor) [inline examples/App.js]: Can you change this file to be `app.js` to match the rest?

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+const path = require('path')
```
Comment (ianstormtaylor) [inline webpack.config.js]: Can you move this file into `./support` like the others?

**Comment by schneidmaster** [comment]
@ianstormtaylor updated!

**Comment by ianstormtaylor** [comment]
Beautiful, thanks!

**Comment by ianstormtaylor** [comment]
@schneidmaster I think this removes the `index.html` that GitHub pages uses?

**Comment by schneidmaster** [comment]
@ianstormtaylor webpack is now generating its own `index.html` via [HtmlWebpackPlugin](https://github.com/ianstormtaylor/slate/blob/master/support/webpack.config.js#L55-L63) (this makes things easier because webpack can then inject its own script tags with the current hash for the JS/CSS)

**Comment by schneidmaster** [comment]
(You should see this in `examples/dist/index.html` after running the production build)

**Comment by ianstormtaylor** [comment]
@schneidmaster ah yup, you're absolutely right.

**Pull request merged**